fastify.js

```diff
@@ -272,7 +272,8 @@ function build (options) {
     this.context = context
   }
 
-  function hookIterator (fn, state, next) {
+  function hookIterator (fn, state, next, release) {
+    if (state.res.finished === true) return release()
```

Comment:
If the hook sends a stream, will `res.finished` be `true` here? You may need to use `res.headersSent` instead.

Reply:
I'll test this tomorrow :)